src/components/RegularButton.vue

```diff
@@ -81,12 +84,53 @@ export default class RegularButton extends Vue {
   @Prop({
     type: Function,
   })
-  readonly onClick!: () => Promise<void>;
+  readonly onClick!: () => Promise<void> | any;
 
-  hover = false;
+  isHover = false;
 
   onHover() {
-    this.hover = !this.hover;
+    this.isHover = !this.isHover;
+  }
+
+  get className() {
+    switch (this.tag) {
+      case 'link':
+        return `base-link base-link_size_${this.size} base-link_type_${this.type} base-link_border-radius-size_${this.border}`;
+      case 'icon':
+        return `base-button base-button-icon_size_${this.size} base-button_type_${this.type} base-button_border-radius-size_${this.border}`;
+      default:
+        return `base-button base-button_size_${this.size} base-button_type_${this.type} base-button_border-radius-size_${this.border}`;
+    }
+  }
+
+  get loaderFill() {
+    return this.type === 'secondary' ||
+      (this.tag === 'link' && this.type !== 'black')
+      ? 'url(#paint0_angular_4802:1871)'
+      : 'url(#paint0_angular_4802:187)';
+  }
+
+  get arrowIconFill() {
+    return this.isDisabled ||
+      (this.tag === 'button' && this.type === 'secondary' && !this.isHover)
+      ? '#6E829A'
+      : this.type === 'secondary' ||
+        (this.tag === 'link' && this.type !== 'black')
+      ? '#0066FF'
+      : 'white';
+  }
```

Comment:
better to set class to element, when in class u define current color, because in real project u will use css or scss variables